### ISSUE TITLE
[🐛 Bug]: Alert closes immediately after calling `window.alert()`

### ISSUE THREAD

**Issue by NearHuscarl** [webdriverio/webdriverio#13610]
### Have you read the Contributing Guidelines on issues?

- [x] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

v9

### Node.js Version

v18.19.1

### Mode

WDIO Testrunner

### Which capabilities are you using?

```typescript
{
  browserName: "chrome",
}
```

### What happened?

Note: This is a separate issue of this one #11832

Calling `window.alert()` on any page in the console will open the alert dialog but close immediately. This only happens in v9. v8 seems to work fine for me.

### What is your expected behavior?

Alert dialog needs to remain open if there is no interaction like calling `browser.acceptAlert()`.

### How to reproduce the bug.

- Clone this repository: https://github.com/webdriverio/jasmine-boilerplate
- Add `browser.debug()` somewhere to stop the test
- Go to any page and open the console
- Run `window.alert('abc')`
- The alert dialog close immediately without any further action.

### Relevant log output

```typescript
cdp:RECV ◂ { "method": "Page.javascriptDialogOpening", "params": { "url": "https://the-internet.herokuapp.com/checkboxes", "message": "sfdsf", "type": "alert", "hasBrowserHandler": true, "defaultPrompt": "" }, "sessionId": "5C513D48DDC9685A4DFFE57C2A0E65F9" }
cdp:SEND ▸ { "id": 23, "method": "Page.handleJavaScriptDialog", "params": { "accept": false }, "sessionId": "5C513D48DDC9685A4DFFE57C2A0E65F9" }
cdp:RECV ◂ { "method": "Page.javascriptDialogClosed", "params": { "result": false, "userInput": "" }, "sessionId": "5C513D48DDC9685A4DFFE57C2A0E65F9" }
cdp:RECV ◂ { "id": 23, "result": {}, "sessionId": "5C513D48DDC9685A4DFFE57C2A0E65F9" }
```
```


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

**Comment by sebastian-sauer** [comment]
Do you have a listener for browser.on('dialog')? 
As far as i understand this is the documented behaviour.

See the blog post: https://webdriver.io/blog/2024/08/15/webdriverio-v9-release#automatic-dialog-handling

And the note at https://webdriver.io/docs/api/dialog/ 

>  Dialogs are dismissed automatically, unless there is a browser.on('dialog') listener. When listener is present, it must either [dialog.accept()](https://webdriver.io/docs/api/dialog/accept) or [dialog.dismiss()](https://webdriver.io/docs/api/dialog/dismiss) the dialog - otherwise the page will freeze waiting for the dialog, and actions like click will never finish.

**Comment by NearHuscarl** [comment]
Thanks for your help. I only checked the release page and missed that blog post completely. For anyone who is interested, to disable dialog suppression by wdio, all you have to do is to subscribe to the `dialog` event to replace the default behavior:

```ts
 beforeAll(() => {
     browser.on("dialog", (dialog) => {});
 });
```

**Comment by jan-molak** [comment]
@NearHuscarl - I just hit the same issue when implementing support for WebdriverIO 9 in Serenity/JS (serenity-js/serenity-js#2572).

The issue seems to be caused by the default `chromium-bidi` alert handler, which defaults dismissing the alert if no other `unhandledPromptBehavior` is specified. See GoogleChromeLabs/chromium-bidi#2556

To handle the dialogs yourself, you can add the following capability to `wdio.conf.ts`
```
capabilities: [{
        browserName: 'chrome',
        unhandledPromptBehavior: 'ignore',
        // ...
}]
```

**Comment by christian-bromann** [comment]
> To handle the dialogs yourself, you can add the `unhandledPromptBehavior` property to your `chrome` capability in `wdio.conf.ts`

This should be set by default in WebdriverIO.

**Comment by christian-bromann** [comment]
This was fixed in https://github.com/webdriverio/webdriverio/pull/13805 and released in `v9.2.2`
